packages/core/src/state-management/__tests__/repository.test.ts

```diff
@@ -758,3 +759,35 @@ describe('handleAnchorEvent', () => {
     })
   })
 })
+
+describe('_registerRunningState', () => {
+  test('deduplicate stream updates', async () => {
+    const state$ = new RunningState(
+      {
+        type: 1,
+        metadata: { controllers: ['did:3:foo'] },
+        content: { a: 1 },
+        anchorStatus: 0,
+        log: [
+          {
+            cid: TestUtils.randomCID(),
+            type: CommitType.GENESIS,
+          },
+        ],
+        signature: 3,
+      },
+      false
+    )
+    const emittedDocuments: Array<FeedDocument> = []
+    const subscription = ceramic.feed.aggregation.documents.subscribe((document) => {
+      emittedDocuments.push(document)
+    })
+    ;(ceramic.repository as any)._registerRunningState(state$)
+    state$.next({
+      ...state$.value,
+      anchorStatus: AnchorStatus.PROCESSING,
+    })
+    expect(emittedDocuments.length).toEqual(1)
```

Comment:
sorry, it's not clear to me how this test is testing the deduplication.  Here I see us emitting one state change and seeing one update to the feed.  I think we'd need to emit two state changes that only affect anchoring status and still see that that only results in one update to the feed.

Reply:
Unless the first update is from the stream creation?  If if you check `emittedDocuments` after registering the running state but before emitting the PROCESSING anchor status, do you also get an emittedDocuments length of 1?  If so, I think duplicating this check to be both before and after we emit the PROCESSING status will make the test clearer.

Reply:
Okay, added more checks to make it clear.